include/cuda/annotated_ptr

```diff
@@ -127,6 +127,7 @@ _Tp* associate_access_property(_Tp* __ptr, _Property __prop) {
     std::is_same<_Property, access_property::persisting>::value ||
     std::is_same<_Property, access_property::streaming>::value ||
     std::is_same<_Property, access_property::normal>::value ||
+    std::is_same<_Property, access_property::global>::value ||
```

Comment:
I just noticed that this PR changes `anotated_ptr`. Was this intentional?

Reply:
Not at all, I merged main without realizing that Wesley had brought in the rollback into main, Need to revert it